src/components/sketches/Scene5_happy.ts

```diff
@@ -0,0 +1,266 @@
+      }
+
+      for (let i = 0; i < this._particles.length; i++) {
+        this._particles[i].show();
+      }
+    }
+
+    explode() {
+      this.exploded = true;
+      for (let i = 0; i < this.explodeSize * EXPLODE_OFFSET; i++) {
+        const particle = new Particle(
+          this._firingParticle.pos.x,
+          this._firingParticle.pos.y,
+          false,
+        );
+        this._particles.push(particle);
+      }
+    }
+  }
+
+  class Particle {
+    pos: p5.Vector;
+    public isFirework: boolean;
+    lifespan: number;
+    private _acc: p5.Vector;
+    rand: number;
+    vel: p5.Vector;
+
+    constructor(x: number, y: number, firework: boolean) {
+      this.pos = p.createVector(x, y);
+      this.isFirework = firework;
+      this.lifespan = FIREWORK_LIFESPAN;
+      this._acc = p.createVector(0, 0);
+      this.rand = p.random(0, 3);
+      this.vel = this.isFirework
+        ? p.createVector(0, p.random(-12, -8))
+        : p5.Vector.random2D().mult(p.random(2, 10));
+    }
+
+    update(force: p5.Vector) {
+      if (!this.isFirework) {
+        this.vel.mult(0.9);
+        this.lifespan -= 4;
+      }
+      this._acc.add(force);
+      this.vel.add(this._acc);
+      this.pos.add(this.vel);
+      this._acc.mult(0);
+    }
+
+    done() {
+      return this.lifespan < 0;
+    }
+
+    show() {
+      if (!this.isFirework) {
+        // after explosion
+        p.strokeWeight(6);
+        p.stroke(255, 211, 97, this.lifespan);
+      } else {
+        p.strokeWeight(8);
+      }
+
+      const opacity = this.rand < 1 ? 0.2 : this.rand < 2 ? 0.4 : 0.6;
+      p.stroke(`rgba(255,211,97,${opacity})`);
+      p.point(this.pos.x, this.pos.y);
+    }
+  }
+
+  class RingParticle {
+    private _pos: p5.Vector;
+    private _vel: p5.Vector;
+    private _acc: p5.Vector;
+    private _radius: number;
+    private _time: number;
+    private _explode: boolean;
+    private _lifespan: number;
+    private _history: p5.Vector[];
+
+    constructor(x: number, y: number, theta: number) {
+      this._pos = p.createVector(x, y); // Position
+      this._vel = p.createVector(2, 2); // Velocity
+      this._radius = RING_RADIUS; //3*noise(this.pos.x/250, this.pos.y/250)*5;
+      this._vel.limit(5);
+      this._acc = p.createVector(0.1, 0.1); // Acceleration
+      this._vel.rotate(theta); // Rotate velocity to the given rotation
+      this._acc.rotate(theta); // Rotate to the given rotation
+      this._time = 0; // Time alive
+      this._explode = false;
+
+      this._lifespan = FIREWORK_LIFESPAN;
+      this._history = [];
+    }
+    public update() {
+      this._vel.add(this._acc);
+      this._pos.add(this._vel); // Update position with velocity
+      // If older than 40, be deleted next draw event
+      if (this._time <= 40) {
+        // If older than 15, reduce the size of the ringParticles every step
+        if (this._time > 15) {
+          this._radius -= 0.05;
+        }
+        this._time++;
```

Comment:
i believe this is possible? 
```suggestion
      if (this._time++ <= 40 && this._time > 15) {
        // If older than 15, reduce the size of the ringParticles every step
        this._radius -= 0.05;
        return;
      }
      ... rest of your logic without the `else` statement
```